### ISSUE TITLE
Why is there a large chromosome area and there is no peak？

### ISSUE THREAD

**Issue by azyro** [LiuLabUB/HMMRATAC#3]
Hi，
    When I use HMMRATAC for ATAC-seq data to call peak,I found there is no peak in the interval of about 10M on a chromosome.Not all chromosomes have this kind of situation。The following figure is the result of my viewing with IGV software.
![no_peak](https://user-images.githubusercontent.com/39055452/39702834-a04e0592-5238-11e8-8a4b-ad8f187791da.jpg)
Hope your replay，thanks .

**Comment by azyro** [comment]
Thanks for your replay！I use -z 60 ,it still have gap ,but when I use the -z 10,it's ok.

**Comment by EvanTarbell** [comment]
Keep in mind that any sites that are masked by HMMRATAC (based on the -z option) are added back to the gappedPeak file and are annotated as "High Coverage Peak".

**Comment by kyliao** [comment]
-z  ,  --zscore <int>  Zscored read depth to mask during Viterbi decoding.
“Zscore” means the average number of reads mapping to the open region minus the genome average divided by the genomic standard deviation？

**Comment by EvanTarbell** [comment]
In this case, the -z option is used to mask regions from the viterbi algorithm, which annotates each genomic position as one of the four states of the model (background, non-flanking nucleosome, flanking nucleosome and open region).  Viterbi has trouble when it encounters a very high coverage region.  Therefore, if the read coverage (denoted by a zscore) is greater than the user defined value of -z, then it masks that region from viterbi and skips over it.  It will then add that skipped region to the resulting peak output file, denoting it as a "High Coverage Peak".  And yes, the zscore is the raw reads mapping to a genomic position minus the genome average and divided by the genomic standard deviation.

**Comment by kyliao** [comment]
In this way，when zscore set small than 100，that should remain more masked region？

**Comment by EvanTarbell** [comment]
Yes, more regions will be masked as you set the -z option lower. These masked regions will be added back to the peak output file.

**Comment by kyliao** [comment]
![image](https://user-images.githubusercontent.com/38950455/41326280-e6b8efd6-6ef0-11e8-813e-ccea81c7daee.png)
But，in this region（zscore = 20），less regions are masked

**Comment by EvanTarbell** [comment]
The figure you are showing doesn't necessarily mean fewer regions are masked.  If you check the output gappedPeak file, you can find all of the masked regions by simply grabbing any line containing the word "High" (ie. grep High FileName_peaks.gappedPeak). The gaps you are seeing aren't the masked regions themselves.  These high coverage regions that we are masking are relatively small.  But what happens is, when HMMRATAC encounters them, it will call the rest of the "block" as either nothing (resulting in these large gaps) or as one large peak.  The "blocks" I'm referring to is this, when HMMRATAC processes the genome with the viterbi algorithm, it splits the genome into "blocks", which are 25MB in size, by default (this can be changed with the --window option).  It assigns a state to every position within that block, then moves on to the next block. It splits the genome into blocks as a memory saving feature.  When it encounters a high coverage region, it can result in the rest of the block being called as a gap.  So when we mask these regions, HMMRATAC skips over them and can continue through the rest of the block.  Hope this clarifies.  
![slide1](https://user-images.githubusercontent.com/5604885/41356772-8ecc3376-6ef2-11e8-8e8d-d7c5107597a7.png)

**Comment by kyliao** [comment]
Thank you for your patience.

**Comment by hmyh1202** [comment]
The output format of  MACS2 include narrowPeak broadPeak and gappedPeak, so which type should I used to further analysis?

**Comment by EvanTarbell** [comment]
@azyro @kyliao 
I am currently in the process of publishing the manuscript for this code and wanted to have some user feedback for the reviewers. I would like to ask you a few questions. 
1) What type of machine did you run HMMRATAC on?
2) How easy was it for you to run the algorithm? 
3) How well have I been in addressing your concerns about using HMMRATAC?
Thank you!
